common/components/TXMetaDataPanel/components/AdvancedGas.tsx

```diff
@@ -101,24 +111,62 @@ class AdvancedGas extends React.Component<Props, State> {
           )}
         </div>
 
-        {dataField && (
-          <div className="AdvancedGas-data">
-            <DataField />
-          </div>
-        )}
-
-        {feeSummary && (
-          <div className="AdvancedGas-fee-summary">
-            <FeeSummary
-              gasPrice={gasPrice}
-              render={({ gasPriceWei, gasLimit, fee, usd }) => (
-                <span>
-                  {gasPriceWei} * {gasLimit} = {fee} {usd && <span>~= ${usd} USD</span>}
-                </span>
-              )}
-            />
-          </div>
-        )}
+        {!scheduling &&
+          dataField && (
+            <div className="AdvancedGas-data">
+              <DataField />
+            </div>
+          )}
+
+        {this.renderFee()}
+      </div>
+    );
+  }
+
+  private renderFee() {
+    const { gasPrice, scheduleGasPrice } = this.props;
+    const { feeSummary } = this.state.options;
+
+    if (!feeSummary) {
+      return;
+    }
+
+    return (
+      <div className="AdvancedGas-fee-summary">
+        <FeeSummary
+          gasPrice={gasPrice}
+          scheduleGasPrice={scheduleGasPrice}
+          render={(data: RenderData) => this.printFeeFormula(data)}
+        />
+      </div>
+    );
+  }
+
+  private printFeeFormula(data: RenderData) {
+    if (this.props.scheduling) {
+      return this.getScheduleFeeFormula(data);
+    }
+
+    return this.getStandardFeeFormula(data);
+  }
+
+  private getStandardFeeFormula({ gasPriceWei, gasLimit, fee, usd }: RenderData) {
+    return (
+      <span>
+        {gasPriceWei} * {gasLimit} = {fee} {usd && <span>~= ${usd} USD</span>}
+      </span>
+    );
+  }
+
+  private getScheduleFeeFormula({ gasPriceWei, scheduleGasLimit, fee, usd }: RenderData) {
+    const { scheduleGasPrice, timeBounty } = this.props;
+
+    return (
+      <div>
+        {timeBounty && timeBounty.value && timeBounty.value.toString()} + {gasPriceWei} *{' '}
```

Comment:
No need for the .toString methods here, they'll get converted to a string regardless

Reply:
Seems they don't get converted implicitly. Removing the .toString methods results in an error:
```javascript
Objects are not valid as a React child (found: object with keys {negative, words, length, red}). If you meant to render a collection of children, use an array instead```